src/LibVLCSharp/Shared/LibVLCOptions.cs

```diff
@@ -0,0 +1,214 @@
+            _options.Add("--no-audio");
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the network caching value in milliseconds.
+        /// </summary>
+        /// <param name="milliseconds">Caching delay in ms (e.g. 300).</param>
+        public LibVLCOptionsBuilder SetNetworkCaching(int milliseconds)
+        {
+            _options.Add($"--network-caching={milliseconds}");
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the file caching value in milliseconds.
+        /// </summary>
+        /// <param name="milliseconds">Caching delay in ms (e.g. 300).</param>
+        public LibVLCOptionsBuilder SetFileCaching(int milliseconds)
+        {
+            _options.Add($"--file-caching={milliseconds}");
+            return this;
+        }
+
+        /// <summary>
+        /// Enables hardware-accelerated decoding using the specified method.
+        /// </summary>
+        /// <param name="method">Decoding method (e.g. dxva2, vaapi, d3d11va).</param>
+        public LibVLCOptionsBuilder EnableHardwareDecoding(string method)
+        {
+            _options.Add($"--avcodec-hw={method}");
+            return this;
+        }
+
+        /// <summary>
+        /// Enables dropping of late video frames.
+        /// </summary>
+        public LibVLCOptionsBuilder DropLateFrames()
+        {
+            _options.Add("--drop-late-frames");
+            return this;
+        }
+
+        /// <summary>
+        /// Enables frame skipping.
+        /// </summary>
+        public LibVLCOptionsBuilder SkipFrames()
+        {
+            _options.Add("--skip-frames");
+            return this;
+        }
+
+        /// <summary>
+        /// Enables or disables use of Xlib.
+        /// </summary>
+        /// <param name="enabled">True to enable Xlib, false to disable.</param>
+        public LibVLCOptionsBuilder UseXlib(bool enabled)
+        {
+            _options.Add(enabled ? "--xlib" : "--no-xlib");
+            return this;
+        }
+
+        /// <summary>
+        /// Adds a custom libvlc option.
+        /// </summary>
+        /// <param name="option">The command-line option to add.</param>
+        public LibVLCOptionsBuilder AddOption(string option)
+        {
+            _options.Add(option);
+            return this;
+        }
+
+        /// <summary>
+        /// Adds multiple custom libvlc options.
+        /// </summary>
+        /// <param name="options">The options to add.</param>
+        public LibVLCOptionsBuilder AddOptions(IEnumerable<string> options)
+        {
+            _options.AddRange(options);
+            return this;
+        }
+
+
+        /// <summary>
+        /// Builds the final LibVLCOptions object including any platform-specific defaults.
+        /// </summary>
+        public LibVLCOptions Build()
+        {
+            var finalOptions = _options;
+
+            if (_includeDebugLogs)
+                finalOptions.Add("--verbose=2");
```

Comment:
What happens when someone writes
```
.EnableDebugLogs().SetVerbosity(1);
```

Might be confusing. I guess it's one or the other? Maybe worth a comment.